internal/provider/resource_cloudflare_tunnel_virtual_network.go

```diff
@@ -0,0 +1,141 @@
+)
+
+func resourceCloudflareTunnelVirtualNetwork() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareTunnelVirtualNetworkSchema(),
+		CreateContext: resourceCloudflareTunnelVirtualNetworkCreate,
+		ReadContext:   resourceCloudflareTunnelVirtualNetworkRead,
+		UpdateContext: resourceCloudflareTunnelVirtualNetworkUpdate,
+		DeleteContext: resourceCloudflareTunnelVirtualNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareTunnelVirtualNetworkImport,
+		},
+		Description: `
+Provides a resource, that manages Cloudflare tunnel virtual networks for Zero Trust. Tunnel
+virtual networks are used for segregation of Tunnel IP Routes via Virtualized Networks to 
+handle overlapping private IPs in your origins.`,
+	}
+}
+
+func resourceCloudflareTunnelVirtualNetworkRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+	tunnelVirtualNetworks, err := client.ListTunnelVirtualNetworks(ctx, cloudflare.TunnelVirtualNetworksListParams{
+		AccountID: accountID,
+		IsDeleted: cloudflare.BoolPtr(false),
+		ID:        d.Id(),
+	})
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to fetch Tunnel Virtual Network: %w", err))
+	}
+
+	if len(tunnelVirtualNetworks) < 1 {
+		tflog.Info(ctx, fmt.Sprintf("Tunnel Virtual Network for ID %s in account %s not found", d.Id(), accountID))
+		d.SetId("")
+		return nil
+	}
+
+	tunnelVirtualNetwork := tunnelVirtualNetworks[0]
+
+	d.Set("name", tunnelVirtualNetwork.Name)
+	d.Set("is_default_network", tunnelVirtualNetwork.IsDefaultNetwork)
+
+	if len(tunnelVirtualNetwork.Comment) > 0 {
+		d.Set("comment", tunnelVirtualNetwork.Comment)
+	}
+
+	return nil
+}
+
+func resourceCloudflareTunnelVirtualNetworkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	name := d.Get("name").(string)
+
+	resource := cloudflare.TunnelVirtualNetworkCreateParams{
+		AccountID: d.Get("account_id").(string),
+		Name:      name,
+		IsDefault: d.Get("is_default_network").(bool),
+	}
+
+	if comment, ok := d.Get("comment").(string); ok {
+		resource.Comment = comment
+	}
+
+	newTunnelVirtualNetwork, err := client.CreateTunnelVirtualNetwork(ctx, resource)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Tunnel Virtual Network %q: %w", name, err))
+	}
+
+	d.SetId(newTunnelVirtualNetwork.ID)
+
+	return resourceCloudflareTunnelVirtualNetworkRead(ctx, d, meta)
+}
+
+func resourceCloudflareTunnelVirtualNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	resource := cloudflare.TunnelVirtualNetworkUpdateParams{
+		AccountID:        d.Get("account_id").(string),
+		Name:             d.Get("name").(string),
+		IsDefaultNetwork: cloudflare.BoolPtr(d.Get("is_default_network").(bool)),
```

Comment:
need to include `VnetID` in the parameters to get the tests passing

```suggestion
		IsDefaultNetwork: cloudflare.BoolPtr(d.Get("is_default_network").(bool)),
		VnetID:           d.Id(),
```